src/irc.nim

```diff
@@ -298,14 +300,41 @@ proc connect*(irc: Irc) =
   assert(irc.port != Port(0))
 
   irc.status = SockConnecting
-  irc.sock.connect(irc.address, irc.port)
+  # check if proxyAddr has been provided
+  if irc.proxyAddr != "":
+    var p = "  "
+    p[0] = cast[char](irc.port.uint16 shr 8)
+    p[1] = cast[char](irc.port)
+
+    # connect to proxy with not authentication
+    irc.sock.connect(irc.proxyAddr, irc.proxyPort)
+    irc.sock.send("\x05\x01\x00") # connect with no auth
+    let resp = irc.sock.recv(2, 1000)
```

Comment:
Where is this defined? Can you link to the spec/RFC? Reading 2 bytes looks suspicious to me.

Reply:
Thats getting the proxy response.